From: Robert Stonehouse <rstonehouse@solarflare.com>
Date: Fri, 2 Mar 2012 17:20:00 +0000
Subject: sfc: Fix calculation of vf_i in map_vi_index()
Patch-mainline: v3.4-rc1
Git-commit: 2c61c8a787fc3382edefd8c7ad3e0f74c81e5302
References: bnc#786035 FATE#314299

This was broken during refactoring to use efx_vf_size().

[bwh: Keep using efx_vf_size()]
Signed-off-by: Ben Hutchings <bhutchings@solarflare.com>
Acked-by: Benjamin Poirier <bpoirier@suse.de>
---
 drivers/net/ethernet/sfc/siena_sriov.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/ethernet/sfc/siena_sriov.c
+++ b/drivers/net/ethernet/sfc/siena_sriov.c
@@ -514,7 +514,7 @@ static bool map_vi_index(struct efx_nic
 
 	if (abs_index < EFX_VI_BASE)
 		return true;
-	vf_i = (abs_index - EFX_VI_BASE) * efx_vf_size(efx);
+	vf_i = (abs_index - EFX_VI_BASE) / efx_vf_size(efx);
 	if (vf_i >= efx->vf_init_count)
 		return true;
 
